src/Tempest/Console/src/Commands/PublishCommand.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console\Commands;
+
+use ReflectionClass;
+use Tempest\Console\Console;
+use Tempest\Console\ConsoleCommand;
+use Tempest\Core\Kernel;
+use Tempest\Support\PathHelper;
+use function Tempest\Support\str;
+use Tempest\Validation\Rules\NotEmpty;
+
+final readonly class PublishCommand
+{
+    public function __construct(
+        private Kernel $kernel,
+        private Console $console
+    ) {
+    }
+
+    #[ConsoleCommand(
+        name: 'publish',
+        description: 'Publish files from vendors'
+    )]
+    public function publish(): void
+    {
+        if (! $this->kernel->publishFiles) {
+            $this->console->error('No files to publish.');
+
+            return;
+        }
+
+        $publish = $this->console->ask(
+            question: 'Which files should be published?',
+            options: $this->kernel->publishFiles,
+            multiple: true,
+            asList: true,
+        );
+
+        foreach ($publish as $file) {
+            $originalPath = (new ReflectionClass($file))->getFileName();
+            $targetPath = $this->console->ask(
+                question: sprintf('Where do you want to publish %s?', $file),
+                default: PathHelper::root($file), // TODO: This doesn't work?
```

Comment:
You could use something like `path($this->kernel->root, '/app')`. But we'd need to store the project's root somewhere (could be `app`, could be `src`, could be something else). Technically this information is available within `Kernel` (`DiscoveryLocations`), but we need to pick the right one.